app/src/main/java/com/cvtracker/vmd/home/MainPresenter.kt

```diff
@@ -227,10 +224,10 @@ class MainPresenter(override val view: MainContract.View) : AbstractCenterPresen
     private fun applyFilters(centers: MutableList<DisplayItem.Center>, filtersList: List<FilterType.FilterSection>){
         filtersList.forEach { section ->
             section.filters.forEach { filter ->
-                if(section.defaultState.not() && filter.enabled) {
+                if(section.defaultState.not() && filter.enabled && filter.predicate != null) {
```

Comment:
Why adding this condition. ? A filter should always reference a predicate, right ?

Reply:
I'm parsing filters using Gson, so function are not kept but rebuild onLoad. The filters are loaded once without function before this re-build.

Reply:
OK je vois.
Donc je pense qu'il y a un petit soucis pour la section "appointments", puisqu'on ne réalimente pas cette section au load, seulement les types vaccins. J'avoue ne pas avoir encore testé cependant :)

Reply:
Oui, c'est ça. Le '@Transcient' permet de sérialisé la liste sans le predicate dans les préférences. Au rappel des préférences le predicate est donc null mais j'ai constaté qu'il est de toute maniére re-alimenter en asynchrone (avec l'ajout de potentiels nouveaux vaccins j'imagine) et la valeure 'enabled' conservé est bien celle des préférences.

J'ai peut-être raté quelque chose, j'aurai bien poser un test pour voir mais pour le coup si il y'a une incidence je ne l'ai pas vu.

Reply:
Pour les types vaccins oui, mais la première section regroupant "Chronodoses uniquement" et "Centres disponibles uniquement", elle n'est pas réalimenté au load.
Du coup je ne crois pas que ça persiste après un Force close de l'app

Reply:
Yes, Bien vu ! Je regarde ça

Reply:
Il y'avait bien un soucis, j'ai fait autrement de fait.